x/gov/legacy/proposals.go

```diff
@@ -0,0 +1,110 @@
+package legacy
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov"
+	"github.com/cosmos/cosmos-sdk/x/gov/proposal"
+)
+
+var cdc *codec.Codec
+
+func init() {
+	cdc = codec.New()
+	cdc.RegisterConcrete(Proposal{}, "gov/TextProposal", nil)
+	cdc.Seal()
+}
+
+type ProposalKind byte
+
+const (
+	ProposalTypeText            ProposalKind = 0x01
+	ProposalTypeSoftwareUpgrade ProposalKind = 0x03
```

Comment:
why `0x03`? shouldn't it be `0x02` ?

Reply:
As I remember `ParameterChangeProposal` occupied `0x02`